storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/TopicBasedRemoteLogMetadataManagerTest.java

```diff
@@ -66,6 +69,28 @@ public TopicBasedRemoteLogMetadataManager topicBasedRlmm() {
         return remoteLogMetadataManagerHarness.remoteLogMetadataManager();
     }
 
+    @Test
+    public void testInternalTopicExists() {
+        Properties adminConfig = remoteLogMetadataManagerHarness.adminClientConfig();
+        ListenerName listenerName = remoteLogMetadataManagerHarness.listenerName();
+        try (Admin admin = remoteLogMetadataManagerHarness.createAdminClient(listenerName, adminConfig)) {
+            String topic = topicBasedRlmm().config().remoteLogMetadataTopicName();
+            boolean doesTopicExist = topicBasedRlmm().doesTopicExist(admin, topic);
+            Assertions.assertTrue(doesTopicExist);
```

Comment:
@kamalcph Can we please verify the flakiness of the test introduced in this PR. I can find the test failing intermittently on PR builds, PR build run which reports this test failing: https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-14632/26/tests/

I have tried to reproduce locally and added `@RepeatedTest(100)` for the test and can reproduce. Can we please either revert this test or fix it.

<img width="1747" alt="Screenshot 2023-11-24 at 5 22 03 PM" src="https://github.com/apache/kafka/assets/2861565/105ae973-7b8e-4f40-9e6a-d34320eb9347">


cc: @satishd

@junrao - Found this while investigating build test failures for PRs in review.

Reply:
Opened #14840 to fix this test. Ran the test locally repeatedly (100) to ensure that it does not fail this time:

<img width="885" alt="Screenshot 2023-11-24 at 11 50 47 PM" src="https://github.com/apache/kafka/assets/11411249/3912af3d-269f-4a34-966f-4a9096b49f30">

PTAL

Reply:
Thanks for fixing this @kamalcph.